packages/aws-cdk-lib/assertions/test/tags.test.ts

```diff
@@ -0,0 +1,95 @@
+import { IConstruct } from 'constructs';
+import {
+  Annotations,
+  App,
+  Aspects,
+  CfnResource,
+  IAspect,
+  Stack,
+} from '../../core';
+import { Tags as _Tags, Match, Tags } from '../lib';
+
+describe('Tags', () => {
+  let app: App;
+
+  beforeEach(() => {
+    app = new App();
+  });
+
+  describe('hasValues', () => {
+    test('simple match', () => {
+      const stack = new Stack(app, 'stack', {
+        tags: { 'tag-one': 'tag-one-value' },
+      });
+      const tags = Tags.fromStack(stack);
+      tags.hasValues({
+        'tag-one': 'tag-one-value',
+      });
+    });
+
+    test('with matchers', () => {
+      const stack = new Stack(app, 'stack', {
+        tags: { 'tag-one': 'tag-one-value' },
+      });
+      const tags = Tags.fromStack(stack);
+      tags.hasValues({
+        'tag-one': Match.anyValue(),
+      });
+    });
+
+    test('no tags with absent matcher will fail', () => {
+      const stack = new Stack(app, 'stack');
+      const tags = Tags.fromStack(stack);
+
+      expect(() => tags.hasValues(Match.absent())).toThrow(
+        /Received \[object Object], but key should be absent/,
```

Comment:
Since the object is defaulted to `{}`, an `absent()` matcher will fail with this message. The assertion just documents this as expected behaviour: if client code uses `Match.absent()` in this fashion, we expect it to fail with this message.

I'm going to add a comment their to make this expectation more clear: I understand why you've questioned it.

In a regex, the `[` character needs to be escaped to be interpreted as a literal instead of starting a character class. The closing `]` doesn't require escaping as its context is not ambiguous; it's treated as a literal.

Reply:
Since I changed the output of `hasValues`, I had to change the matcher, so this troublesome regex is gone 😁